docs/cache/how-to/rate-limiter.mdx

```diff
@@ -0,0 +1,53 @@
+
+Some common use-cases of rate-limiting includes:
+
+- API Management: In a platform offering various APIs, rate-limiting is crucial to prevent a single user or service from monopolizing the bandwidth, ensuring that all users have equitable access to the resources.
+
+- E-commerce Websites: During high-traffic events like Black Friday sales, rate-limiting can prevent the website from crashing by controlling the influx of user requests, thus providing a stable and fair shopping experience to all customers.
+
+- Online Gaming Servers: Rate-limiting can help in mitigating cheating by throttling the number of actions a player can perform in a given time, ensuring a level playing field and maintaining the game's integrity.
+
+## Using Momento to build a distributed rate-limiter
+
+Let’s imagine you want to create a distributed rate-limiter that could effectively manage transactions-per-minute (TPM) for individual users.
+Our approach utilizes Momento's `increment` and `updateTTL` [APIs](./../develop/api-reference). This method proves to be not only efficient but also highly accurate.
+
+At the heart of our rate-limiter is a key mechanism that allows us to perform rate limiting based on user-per-minute granularity. The key is constructed using a combination of a user or entity and the current minute. This key plays a pivotal role in tracking and limiting the number of transactions a user can make in a given minute.
+
+The rate limiter increments the value of the unique key for each user when they make a request, setting a time-to-live (TTL) for the first request of the minute to 60 seconds. This is important as we want our keys to expire as they are not meaningful after they have served their purpose for a given minute.
+
+A flow of the rate-limiter looks like:
+
+- Increment the value of `user_id-current_minute`. If the returned value is 1, that indicates that this was the first request for the user for that given minute. Note that Momento's `increment` API is guaranteed to be atomic. If this return value is 1, we set the TTL of that key using `updateTTL` API to be 60 seconds.
+- If the value is less than the configured TPM limit for the rate limiter, we allow the request, or else, throttle it.
+
+Let's dive right into our implementation; pay attention to comments in this code where we explain the thought process.
+
+<SdkExampleCodeBlock language={'javascript'} file={'examples/nodejs/rate-limiter/doc-examples-files/doc-examples.ts'} />
```

Comment:
We need to plug tabs here. Not blocking right now but will be when we plan to add a new language. I played around the code and need some React knowledge to successfully get it out so will do that on my own time over the next few days/week https://github.com/momentohq/public-dev-docs/issues/506 have assigned myself to it.

Reply:
You can use the `SdkExampleTabs` component here @pratik151192. Take a look at `./docs/cache/develop/api-reference/index.mdx`

Reply:
yea that  just supports functions so we need to add some react stuff to support files + tabs https://gomomento.slack.com/archives/C055MCGD2CD/p1698969486257659